packages/field-base/src/input-field-mixin.js

```diff
@@ -0,0 +1,268 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { Debouncer } from '@polymer/polymer/lib/utils/debounce.js';
+import { animationFrame } from '@polymer/polymer/lib/utils/async.js';
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+import { ClearButtonMixin } from './clear-button-mixin.js';
+import { DelegateFocusMixin } from './delegate-focus-mixin.js';
+import { FieldAriaMixin } from './field-aria-mixin.js';
+import { InputPropsMixin } from './input-props-mixin.js';
+
+const InputFieldMixinImplementation = (superclass) =>
+  class InputFieldMixinClass extends ClearButtonMixin(FieldAriaMixin(InputPropsMixin(DelegateFocusMixin(superclass)))) {
+    static get properties() {
+      return {
+        /**
+         * Whether the value of the control can be automatically completed by the browser.
+         * List of available options at:
+         * https://developer.mozilla.org/en/docs/Web/HTML/Element/input#attr-autocomplete
+         */
+        autocomplete: {
+          type: String
+        },
+
+        /**
+         * This is a property supported by Safari that is used to control whether
+         * autocorrection should be enabled when the user is entering/editing the text.
+         * Possible values are:
+         * on: Enable autocorrection.
+         * off: Disable autocorrection.
+         */
+        autocorrect: {
+          type: String
+        },
+
+        /**
+         * This is a property supported by Safari and Chrome that is used to control whether
+         * autocapitalization should be enabled when the user is entering/editing the text.
+         * Possible values are:
+         * characters: Characters capitalization.
+         * words: Words capitalization.
+         * sentences: Sentences capitalization.
+         * none: No capitalization.
+         */
+        autocapitalize: {
+          type: String
+        },
+
+        /**
+         * Specify that the value should be automatically selected when the field gains focus.
+         */
+        autoselect: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * The value of the field.
+         */
+        value: {
+          type: String,
+          value: '',
+          observer: '_valueChanged',
+          notify: true
+        }
+      };
+    }
+
+    static get hostProps() {
+      return [...super.hostProps, 'autocapitalize', 'autocomplete', 'autocorrect'];
+    }
+
+    static get observers() {
+      return ['__observeOffsetHeight(errorMessage, invalid, label, helperText)'];
+    }
+
+    /** @protected */
+    get _ariaTarget() {
+      return this._inputNode;
+    }
+
+    /**
+     * @return {HTMLElement | undefined}}
+     * @protected
+     */
+    get inputElement() {
```

Comment:
Some places in this file use `_inputNode` directly, others get it through ` get inputElement()`. I can't tell if that's intentional, especially since their names are very similar.

Reply:
Removed this getter and updated the code to use `_inputNode` consistently.